pennylane/transforms/core/transform_dispatcher.py

```diff
@@ -229,6 +234,55 @@ def qfunc_transformed(*args, **kwargs):
 
         return qfunc_transformed
 
+    def _old_device_transform(self, original_device, targs, tkwargs):
+        """Apply the transform on a device"""
+        new_dev = copy.deepcopy(original_device)
+        transform = self._transform
+
+        @new_dev.custom_expand
+        def new_expand_fn(self, tape, *args, **kwargs):  # pylint: disable=unused-variable
+            tapes, _ = transform(tape, *targs, **tkwargs)
+            tape = tapes[0]
+            return self.default_expand_fn(tape, *args, **kwargs)
+
+        return new_dev
+
+    def _device_transform(self, original_device, targs, tkwargs):
+        """Apply the transform on a device"""
+        if self._expand_transform:
+            raise TransformError("Device transform does not support expand transforms.")
+        if self._is_informative:
+            raise TransformError("Device transform does not support informative transforms.")
+        if self._final_transform:
+            raise TransformError("Device transform does not support final transforms.")
+
+        class TransformedDevice(type(original_device)):
+            """A transformed device with updated preprocess method."""
+
+            def __init__(self, original_device, transform):
+                for key, value in original_device.__dict__.items():
+                    self.__setattr__(key, value)
```

Comment:
is this a standard way to copy an unknown object? feels kinda spooky

Reply:
Yes kinda python black magic, but its black magic that gets the job done for an edge case.

Reply:
I agree, I tried to find something else but nothing was cleaner.